lib/TuyaOAuth2App.js

```diff
@@ -1,110 +1,125 @@
-'use strict';
+"use strict";
 
-const { OAuth2App } = require('homey-oauth2app');
-const TuyaOAuth2Client = require('./TuyaOAuth2Client');
+const { OAuth2App } = require("homey-oauth2app");
+const TuyaOAuth2Client = require("./TuyaOAuth2Client");
+const TuyaOAuth2Util = require("./TuyaOAuth2Util");
 
 class TuyaOAuth2App extends OAuth2App {
-
   static OAUTH2_CLIENT = TuyaOAuth2Client;
-  static OAUTH2_DEBUG = process.env.DEBUG === '1';
+  static OAUTH2_DEBUG = process.env.DEBUG === "1";
   static OAUTH2_MULTI_SESSION = false; // TODO: Enable this feature & make nice pairing UI
 
   async onOAuth2Init() {
     await super.onOAuth2Init();
 
-    // Register Tuya Web API Flow Cards
-    this.homey.flow
-      .getActionCard('send_command_string')
-      .registerRunListener(async ({ device, code, value }) => {
-        if (typeof code === 'object') code = code.id;
+    const sendCommandRunListener = async ({ device, code, value }) => {
+      if (typeof code === "object") code = code.id;
+      await device.sendCommand({ code, value });
+    };
 
-        await device.sendCommand({ code, value });
-      })
-      .registerArgumentAutocompleteListener('code', async (query, args) => {
-        const status = await args.device.getStatus();
-        return status
-          .filter(({ value }) => {
-            return typeof value === 'string' && !value.startsWith('{');
-          })
-          .filter(({ code }) => {
-            return code.toLowerCase().includes(query.toLowerCase());
-          })
-          .map(({ code }) => ({
-            id: code,
-            title: code,
-          }));
-      });
+    const generalControlAutocompleteListener = async (query, args, filter) => {
+      const status = await args.device.getStatus();
+      return status
+        .filter(filter)
+        .filter(({ code }) => {
+          return code.toLowerCase().includes(query.toLowerCase());
+        })
+        .map(({ code }) => ({
+          id: code,
+          title: code,
+        }));
+    };
 
+    // Register Tuya Web API Flow Cards
+    // Sending
     this.homey.flow
-      .getActionCard('send_command_number')
-      .registerRunListener(async ({ device, code, value }) => {
-        if (typeof code === 'object') code = code.id;
-
-        await device.sendCommand({ code, value });
-      })
-      .registerArgumentAutocompleteListener('code', async (query, args) => {
-        const status = await args.device.getStatus();
-        return status
-          .filter(({ value }) => {
-            return typeof value === 'number';
-          })
-          .filter(({ code }) => {
-            return code.toLowerCase().includes(query.toLowerCase());
-          })
-          .map(({ code }) => ({
-            id: code,
-            title: code,
-          }));
-      });
+      .getActionCard("send_command_string")
+      .registerRunListener(async (args) => sendCommandRunListener(args))
```

Comment:
```suggestion
      .registerRunListener(sendCommandRunListener)
```

I do not believe the additional function wrapper is necessary here.

Reply:
Fixed in 6619eccc011049297b03d5f28ade46174224074c